### PR TITLE
Global search

### PR DESCRIPTION
**Pull request by andy250**
- global search available on Reddit instance
- possibility to search for posts about a specific URL

### CODE REVIEW

File: RedditSharp/Reddit.cs [hunk 1]
```diff
@@ -317,6 +319,17 @@ public Comment GetComment(string subreddit, string name, string linkName)
             }
         }
 
+        public Listing<T> SearchByUrl<T>(string url) where T : Thing
+        {
+            var urlSearchQuery = string.Format(UrlSearchPattern, url);
+            return Search<T>(urlSearchQuery);
+        }
+
+        public Listing<T> Search<T>(string query) where T : Thing
```
Comment (lufivuahaitac) [inline RedditSharp/Reddit.cs]: Can you give me a specific example for this statement? Eg: the keyword is "beautiful". Thanks

Comment (andy250) [inline RedditSharp/Reddit.cs]: Not sure exactly what you need, but this should work just fine for a simple search:
var posts = new Reddit().Search<Post>("beautiful").GetLIsting().ToList();

**Comment by lufivuahaitac** [comment]
image
When i use your code. I get this error.

**Comment by SirCmpwn** [comment]
`Search<Post>(...`

**Comment by lufivuahaitac** [comment]
@SirCmpwn  thanks you bro :+1:

**Pull request merged**